### ISSUE TITLE
EtcdHighNumberOfFailedGRPCRequests 

### ISSUE THREAD

**Issue by nabbdl** [openshift/cluster-monitoring-operator#248]
I enabled "cluster monitoring operator" on several OKD 3.11 clusters. Everything is working fine except for ETCD monitoring. I followed the documentation to enable etcd monitoring. It's seems to work : most of the checks are green, except the "EtcdHighNumberOfFailedGRPCRequests" which is always triggered  (etcd cluster is working correctly). Do I miss something or is there any know issue while enabling etcd cluster monitoring ?
![image](https://user-images.githubusercontent.com/32450290/52784094-42589b00-3054-11e9-94b7-e24040761fb2.png)

**Comment by nabbdl** [comment]
To be more specific the query regarding this alert always retrieve "**100**" as a value. I suspect that the query is false.
**Current query is** :  
100 * sum by(job, instance, grpc_service, grpc_method) (rate(grpc_server_handled_total{grpc_code!="OK",job=~".*etcd.*"}[5m])) / sum by(job, instance, grpc_service, grpc_method) (rate(grpc_server_handled_total{job=~".*etcd.*"}[5m])) > 5

**in my opinion it should be** : 
sum by(job, instance, grpc_service, grpc_method) (rate(grpc_server_handled_total{grpc_code!="OK",job=~".*etcd.*"}[5m])) / sum by(job, instance, grpc_service, grpc_method) (rate(grpc_server_handled_total{job=~".*etcd.*"}[5m])) > 5

**Comment by benhwebster** [comment]
I also see this on OpenShift 3.11 after enabling etcd monitoring.

**Comment by metalmatze** [comment]
The current query seems correct to me. It alerts on more than 5% of requests failing.
On my personal cluster I can see that there's a watch method pending for 4min.

```
{grpc_method="Watch",grpc_service="etcdserverpb.Watch",instance="10.135.73.45",job="etcd"}
```

Is it the same for you? Maybe we should ignore the watches here.

**Comment by nabbdl** [comment]
Tested your query, result is empty. For me the strange thing is that I have the same alert on all the OpenShifts Cluster I have installed. and the query result for EtcdHighNumberOfFailedGRPCRequests  gives alway a value of 100 
![image](https://user-images.githubusercontent.com/32450290/52859810-00038c80-312e-11e9-8e02-13e4189ef65c.png)

**Comment by zot24** [comment]
Hi @nabbdl just wondering if you still seeing that error and if you resolve the mystery? I'm facing the same error here and not sure why is happening.

**Comment by nabbdl** [comment]
Hi @zot24. Unfortunately, I'm still seeing the same error.

**Comment by zot24** [comment]
@nabbdl jtlyk I have been doing some research and after reading a lot of comments I think I'll just ignore those alert for now https://github.com/poseidon/typhoon/pull/175 there are a bunch of issues regarding this error message, but pretty much what's going on it's summarized in this issue https://github.com/etcd-io/etcd/issues/10289 and there is still not a fix for it

In more detail I think this is the offended line https://github.com/gyuho/etcd/blob/0cf9382024da6132cb5f0778c3fb43e4a6c88afd/etcdserver/api/v3rpc/util.go#L111

**Comment by zot24** [comment]
If you using jsonnet you could add the following to suppress that rule for now:

```
{
  prometheusAlerts+:: {
    groups: std.map(
      function(group)
        if group.name == 'etcd' then
          group {
            rules: std.filter(
              function(rule)
                rule.alert != 'etcdHighNumberOfFailedGRPCRequests',
              group.rules
            ),
          }
        else
          group,
      super.groups
    ),
  },
}

```

**Comment by zot24** [comment]
@nabbdl jtlyk this just got merge https://github.com/openshift/cluster-monitoring-operator/pull/340

**Comment by nabbdl** [comment]
@jtlyk thank you for the update. I’m currently using the « cluster monitoring operator » provided with OpenShift so I can’t use jsonnet to disable the rule. The only thing I can do for now is to completely disable « etcd monitoring ». Or maybe the cluster-monitoring operator will update itself and will take into account the modification ??

**Comment by benhwebster** [comment]
You could do what I did in my clusters and create a silence for those alerts in alertmanager, but it does look like they may be backporting it currently: https://github.com/openshift/cluster-monitoring-operator/pull/383

**Comment by paulfantom** [comment]
Fix backported in #383 

/close
